commit d83247c03d05bd386bd62ba4aee5f0f6725a9937
Author: Nils Philippsen <nils@redhat.com>
Date:   Tue Jun 16 17:03:12 2009 +0200

    patch: hal
    
    Squashed commit of the following:
    
    commit a50c619ddbb869612e99a9457b1f39492f34b141
    Author: Nils Philippsen <nils@redhat.com>
    Date:   Tue Jun 16 15:21:59 2009 +0200
    
        adapt HAL fdi files for Fedora

diff --git a/tools/Makefile.am b/tools/Makefile.am
index bd6dd9d..bb26fcb 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -79,7 +79,7 @@ udev/libsane.rules: $(wildcard ${top_srcdir}/doc/descriptions/*.desc) $(wildcard
 	   -d 0 > $@
 
 hal/libsane.fdi: $(wildcard ${top_srcdir}/doc/descriptions/*.desc) $(wildcard ${top_srcdir}/doc/descriptions-external/*.desc) sane-desc
-	@./sane-desc -m hal -s ${top_srcdir}/doc/descriptions:${top_srcdir}/doc/descriptions-external \
+	@./sane-desc -m hal-new -s ${top_srcdir}/doc/descriptions:${top_srcdir}/doc/descriptions-external \
 	   -d 0 > $@
 
 sane-backends.pc:	$(top_builddir)/config.status
diff --git a/tools/sane-desc.c b/tools/sane-desc.c
index 600ab03..db46806 100644
--- a/tools/sane-desc.c
+++ b/tools/sane-desc.c
@@ -3667,7 +3667,6 @@ print_hal (int new)
       printf (" -->\n");
       printf ("        <match key=\"usb.product_id\" int=\"%s\">\n", usbid->usb_product_id);
       printf ("          <append key=\"info.capabilities\" type=\"strlist\">scanner</append>\n");
-      printf ("          <merge key=\"scanner.access_method\" type=\"string\">proprietary</merge>\n");
       printf ("        </match>\n");
 
       usbid = usbid->next;
@@ -3676,6 +3675,11 @@ print_hal (int new)
   if (in_match)
     printf ("      </match>\n");
 
+  printf ("\n      <!-- The following rule will disable USB autosuspend for USB scanner devices -->\n");
+  printf ("      <match key=\"info.capabilities\" contains=\"scanner\">\n");
+  printf ("        <append key=\"info.callouts.add\" type=\"strlist\">hal-usb-scanner-add</append>\n");
+  printf ("      </match>\n");
+
   printf ("    </match>\n");
 
   printf ("  </device>\n");
